packages/core/src/common/browserSupport.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+export function elementClosest(element: Element, selector: string): Element | null {
+    let ancestor: Element | null = element;
+    while (ancestor !== null && !ancestor.matches(selector)) {
+        ancestor = element.parentElement;
+    }
+
+    return ancestor;
+}
+
+export function addDomTokenListItems(classList: DOMTokenList, classes: string[]) {
```

Comment:
i think this is only used once. might prefer inlining it as a `.split(" ").map()`. thoughts?